Rename beta SDKs to public preview SDKs in readme (#3126)
stripe/stripe-dotnet: README.md

StripeConfiguration.EnableTelemetry = false;
```

### Public Preview SDKs

> Stripe has features in the [public preview phase](https://docs.stripe.com/release-phases) that can be accessed via the beta version of this package.

Reviewer: a small nit but "the beta version of this package" reads strangely given we also refer to the "public preview SDKs".  Maybe replace this with "a specific set of versions of this package" or something along those lines?

Author: should be better with https://github.com/stripe/stripe-dotnet/pull/3126/commits/7b5af001d5e30505d520bdd266d6ea202ca934d8

We would love for you to try these as we incrementally release new features and improve them based on your feedback.

> The public preview SDKs are just a different version of the same package and are appended with `-beta.X` such as `45.0.0-beta.1`. To install, choose the version that includes support for the preview feature you are interested in by reviewing the [releases page](https://github.com/stripe/stripe-dotnet/releases/) and then use it in the version parameter with `dotnet add package` command:

Author: ```suggestion
The public preview SDKs are different versions of the same package as the stable SDKs. These versions are appended with `-beta.X` such as `45.0.0-beta.1`. To install, choose the version that includes support for the preview feature you are interested in by reviewing the [releases page](https://github.com/stripe/stripe-dotnet/releases/) and then use it in the version parameter with `dotnet add package` command:
```

```
dotnet add package Stripe.net --version <replace-with-the-version-of-your-choice>
```

> **Note**
> There can be breaking changes between two versions of the public preview SDKs without a bump in the major version. Therefore we recommend pinning the package version to a specific version in your project file. This way you can install the same version each time without breaking changes unless you are intentionally looking for the latest public preview SDK.

> If your beta feature requires a `Stripe-Version` header to be sent, set the `StripeConfiguration.ApiVersion` property with the `StripeConfiguration.AddBetaVersion` function (available only in the public preview SDKs):

Reviewer: maybe outside the scope of this PR, but I first read "set the `StripeConfiguration.ApiVersion` property" as an instruction to try and set that property.  perhaps: "use `StripeConfiguration.AddBetaVersion` method to set the api version (accessed by the `StripeConfiguration.ApiVersion` property)" or something like that?

Author: should be better with https://github.com/stripe/stripe-dotnet/pull/3126/commits/7b5af001d5e30505d520bdd266d6ea202ca934d8

Reviewer: yea this is better
